src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,169 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null && tail == null) {
+            head = new Node<T>(null, value, null);
+            tail = head;
+            size++;
+        } else {
+            head.next = new Node<T>(head, value, null);
```

Comment:
When creating a new Node<T>, pass all the fields to the Node constructor. Here, you should also update the tail, not the head, as you're adding to the end of the list.